src/utils/generateNotificationData.ts

```diff
@@ -24,7 +24,7 @@ class NotificationData {
         const axlesSeats = (event.activity.vehicleType === VEHICLE_TYPES.PSV) ? event.activity.numberOfSeats : event.activity.noOfAxles;
         const vrmTrailerId = (event.activity.vehicleType === VEHICLE_TYPES.TRL) ? event.activity.trailerId : event.activity.vrm;
         personalization.activityDetails += `^#${this.capitalise(event.activityType)} (${vrmTrailerId})
-      ^• Time: ${this.formatDateAndTime(event.activity.testTypes.testTypeStartTimestamp, "time")} - ${this.formatDateAndTime(event.activity.testTypes.testTypeEndTimestamp, "time")}
+      ^• Time: ${this.formatDateAndTime(event.activity.testStartTimestamp, "time")} - ${this.formatDateAndTime(event.activity.testEndTimestamp, "time")}
```

Comment:
I think I will need to have a catch up to understand the event better. My understanding is that  the even returns a list of activities and I can see that we have the following keys (from activity services depending on the type of visit) `startTime` and `endTime`.

Reply:
the activity object here is either an activity or a test result object, with the `IActivitiesList.activityType` indicating which.  So at this point, we are just assuming that the activity object is a test result, and has these attributes.  Which tbf it should.

Using` testStartTimestamp`/ and `testEndTimestamp` aligns to the retro key report, so it explains why this fixes the issue.

Reply:
Thank you 👍

Reply:
one for the future, we could use object destructuring to make it more concise like that
```
const {activity: {testStartTimestamp, testEndTimestamp}} = event

// ...
formatDateAndTime(testStartTimestamp)
```

so we make it more concise and easier to read in the function call.